### DISCUSSION TITLE
TextBox after Undo CaretIndex forwards always

### DISCUSSION THREAD

**Discussion by postrondev** [AvaloniaUI/Avalonia#14328] [Q&A]
i have a Textbox_TextChanging event handler

if ((Text == "1239456" ) && (CaretIndex == 3))
{
disable TextChanging
Undo
enable TextChanging
}

i enter 1 2 3 4 5 6 
go to CaretIndex 3
i enter 9

the text goes back to 123456 and CaretIndex forwards to 4
the CaretIndex should return to 3

**Comment by TETYYS** [comment]
Your code is not valid C# code, it is not clear what `Undo` does

**Comment by timunie** [comment]
@postrondev please file a minimal sample and fill out issue forms. If it is a question, please use discussions. I'll convert this into a discussion as I think something is wrong on your side.
